ext/socket/raddrinfo.c

```diff
@@ -9,6 +9,7 @@
 ************************************************/
 
 #include "rubysocket.h"
+#include <pthread.h>
```

Comment:
Do we need to handle other platforms that don't have pthread?

Reply:
Seems like this breaks on Windows which is not a surprise. Should I wrap it into `#ifdef __linux__`?

Reply:
You should find a way to make sure this works for all platforms Ruby currently support for this method.  Ideally, it would work the same on all platforms.  If that isn't possible, you should fall back to the previous implementation for platforms that cannot be handled with the new implementation.

This approach seems like it would work with any platform that supports pthread, so maybe `#if defined(HAVE_PTHREAD_H)` (though I'm not sure if that is exposed to this file as it is to thread.c).